### DISCUSSION TITLE
how to specify the OCR of 4 digits (0000~9999)

### DISCUSSION THREAD

**Discussion by DidibabaQiu** [hertzg/tesseract-server#75] [General]
Dear  sir, 

I am a novice, could you please teach me how to specify the OCR of 4 digits (0000~9999), or where should I go to find the answer? 

here is my command (home-assistant):
`command: curl -F "options={\"languages\":[\"eng\"], \"ocrEngineMode\":0}" -F file=@"my jpg filename" http://192.168.1.1:8884/tesseract`

The ocr result is often "i234" or "123o" or "7B9o" or non-numeric English characters. By the way, I'm going to use it on my gas meter.

Thank you sir for any response.

**Comment by hertzg** [comment]
This question is not related to this service and is specific to tesseract, but just because this specific case matches one of mine I can share some insights:

Tesseract has a lot of configuration options (see the configuration options available for it) you might want to experiment with them and find what matches yours best.


Here's a sample gas meter I found online.

![image](https://user-images.githubusercontent.com/1886698/223959443-65a93a01-944c-4a74-8bb2-6ce3e911ae6b.png)

I just cut the digits part as a full line (not the best approach but enough for the example)

![digits](https://user-images.githubusercontent.com/1886698/223959253-d324a062-7bf2-44f6-a5ca-53a2f4eb2df2.jpg)

```json
{
  "languages": ["eng"],
  "ocrEngineMode": 0,
  "pageSegmentationMethod": 6,
  "dpi": 70,
  "configParams": {
    "load_system_dawg": "false",
    "load_freq_dawg": "false",
    "tessedit_char_whitelist": "0123456789",
    "page_separator": ""
  }
}
```

yields

```json
{
  "data": {
    "exit": {
      "code": 0,
      "signal": null
    },
    "stderr": "",
    "stdout": "10 0 0   1 4  3   9 6 1\n"
  }
}
```

as you can see the results are not the best, the first `1` is probably because of the way I cropped the image, there is a small whilte section there which is interpreted as `1`. Probably the same applies to the last `1` following the `6`.

You will get much better results if you cut out each segment as a spearate image with prominent contrast between the background color and the digit color.

**Reply by DidibabaQiu** [reply]
Mr. hertzg, 

Thank you very much for your help, it's a great help for me.
